### PR TITLE
Course slides improvements

### LINKED ISSUE #4: Envrc on linux
There is a `.envrc` file at the root of the directory (for Windows user I suppose?), that conflicts with the need of an `.envrc` file containing `use nix` to use `direnv` on linux/macos.

Is there something we can do about that?

### PR DESCRIPTION
**Pull request by tbagrel1**
Fixes #4

### CODE REVIEW

File: slides/course-04.md [hunk 1]
```diff
@@ -226,7 +226,28 @@ interface Addable<A> {
...
 # Typeclasses vs Java interfaces
 
-Since there is no subtyping, TODO @smelc
+Since there is no subtyping (and no concept of _trait object_ or
+_interface object_), we can't build nor work on heterogeneous collections.
+Compare:
+
+[//]: #exdown-skip
+```hs
+-- Can sum a list of @a@s when @a@ has the 'Num' capability.
```
Comment (smelc) [inline slides/course-04.md]:
```suggestion
-- | Can sum a list of @a@s when @a@ has the 'Num' capability.
```

Let's show the best practices to the students :-)

File: slides/course-04.md [hunk 2]
```diff
@@ -226,7 +226,28 @@ interface Addable<A> {
...
+```
+
+[//]: #exdown-skip
+```java
+interface Num { ... }
+class NInt implements Num { ... }
+class NDouble implements Num { ... }
+// 'nums' can contain some items of concrete type NInt,
+// and some items of concrete type NDouble at the same time.
+// The returned value can be either a NInt or NDouble.
```
Comment (smelc) [inline slides/course-04.md]:
```suggestion
/** 'nums' can contain some items of concrete type NInt,
  * and some items of concrete type NDouble at the same time.
  * The returned value can be either a NInt or NDouble. */
```

File: slides/course-04.md (hunk 1 above)
Comment (tbagrel1) [reply slides/course-04.md]: This is what happens when I try to do Haddock without looking at the doc :sweat_smile:

File: .envrc [hunk 3]
```diff
@@ -1,3 +1 @@
-# requires ghcup to be in $PATH
-PATH_add $(pwd)/ghc/bin # Populate with: ghcup install ghc 8.10.7 --isolate $(pwd)/ghc
-PATH_add $(pwd)/hls/bin # Populate with: ghcup install hls  --isolate $(pwd)/hlc
+use nix
```
Comment (smelc) [inline .envrc:1]:
```suggestion
use nix
source_env_if_exists .envrc.private # allow custom extensions
```

**Review by smelc — APPROVED** [review]

**Pull request merged**